### PR TITLE
add logging via pino to apps/web

### PR DESCRIPTION
**Pull request by brpins**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/web/src/server/lib/logger.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import config from '@pins/web/environment/config.js';
+import path from 'node:path';
+import pino from 'pino';
+
+/** @type {import('pino').TransportTargetOptions[]} */
+const targets = [];
+
+if (config.isProd) {
+	targets.push({
```
Comment (radum) [inline apps/web/src/server/lib/logger.js:9]: I think we need the pretty version here also, but maybe with a different level. Just to make sure the logstream has the info we need when we look at it.

File: apps/web/src/server/lib/logger.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		target: 'pino/file',
+		level: 'trace',
```
Comment (radum) [inline apps/web/src/server/lib/logger.js:11]: Shouldn't this come from the env vars?

File: apps/web/src/server/lib/logger.js (hunk 1 above)
Comment (brpins) [reply apps/web/src/server/lib/logger.js:9]: so you want to pretty-print before it's written into the log file? I'd have thought it's better to save them in a one-log-per-line for a log file? (bear in mind they can still be pretty printed from the log file if needed)

File: apps/web/src/server/lib/logger.js (hunk 2 above)
Comment (brpins) [reply apps/web/src/server/lib/logger.js:11]: It's not really necessary, as at the top level of the pino instantiation, the log level is passed from the `config`, which means logs won't be captured for anything lower level than the env var. Putting `trace` in the transport is basically saying "we're deferring to the master log level".

File: apps/web/src/server/lib/logger.js (hunk 1 above)
Comment (radum) [reply apps/web/src/server/lib/logger.js:9]: I would say debug for file and info for console. But I think both should be configured separately.

Comment (radum) [reply apps/web/src/server/lib/logger.js:9]: Good point, I wonder if we should replace morgan with https://github.com/pinojs/pino-http to be consistent with the tool and pretty that one to console only for now.

Comment (brpins) [reply apps/web/src/server/lib/logger.js]: Have now updated the PR with a change to transport construction. Regarding pino-http, I'll try it.

**Comment by radum** [comment]
LGTM

**Comment by brpins** [comment]
@radum can you re-approve? I had to resolve merge conflicts

**Pull request merged**